gomock/call_test.go

```diff
@@ -0,0 +1,47 @@
+package gomock
+
+import "testing"
+
+type mockTestReporter struct {
```

Comment:
I could have generated a mock for the `TestReporter` interface but unfortunately in our case this isn't possible:

Since the generated mocks import to gomock package we have to put the mock to a separate package (this is why we have the `gomock/mock_matcher` package) and the gomock package can't import that package because that would cause a circular dependency. This could still be avoided by putting our tests to the gomock_test package (blackbox testing - this is how matchers_test.go uses the mock_matcher package) but we have to keep the tests in the gomock package (whitebox testing) because we have to access the unexported `Call.t` field.